### PR TITLE
#168050017 Add Search Functionality

### PR DESCRIPTION
**Pull request by gik-hub**
#### What does this PR do?
Adds search functionality
#### Description of Task to be completed?
- Add endpoint to search the requests table  by a keyword or by bate
#### How should this be manually tested?
- git clone this repo and check out to `ft-add-request-search-functionality-168050017`
- run `npm i` to install all the dependencies
- run the application with `npm run dev`
- login a user in the system
- send a GET request to `/api/v1/requests/search?query`
     * In the query options either use:
         ` key_word , afterDate, beforeDate` as search options
#### Images
image

#### Any background context you want to provide?
N/A
#### What are the relevant pivotal tracker stories?
[#168050017](https://www.pivotaltracker.com/story/show/168050017)
#### Questions:
N/A

### CODE REVIEW

**Review by Jordybastien — APPROVED** [review]
Good work Titus

**Review by Jordybastien — APPROVED** [review]
Good work @TgeeT, However if you could work on the issue that is decreasing our coverage that would be great

**Comment by gik-hub** [comment]
@Jordybastien, I really appreciate the feedback, I have worked on the coverage and it is now fine

File: src/routes/index.js [hunk 1]
```diff
@@ -5,13 +5,15 @@ import swaggerDoc from '../../swagger.json';
 import AuthRoutes from './AuthRoutes';
 import UserRoutes from './UserRoutes';
 import AdminRoute from './AdminRoutes';
+import RequestRoutes from './RequestRoutes';
 import AccommodationRoutes from './AccomodationsRoutes';
 import RequestRoute from './RequestRoutes';
```
Comment (didasmbalanya) [inline src/routes/index.js]: request route has been imported twice

**Review by amilykassim — CHANGES_REQUESTED** [review]
The functionality is working perfectly, however there 2 main issues : 
1)  when querying the data before
the current date and it is ending upbringing the data that is after the current date instead of the ones before the current date.

2) When querying the data against the range of dates it is indicating that there are no data in that range even if the data exist.

Otherwise, Great work, Kudos to you Titus, the search is awesome!!

**Comment by gik-hub** [comment]
@amilykassim I have fixed the same, it was a matter of logic flow, it is now working appropriately

**Review by didasmbalanya — APPROVED** [review]

**Review by amilykassim — APPROVED** [review]

**Review by Paulvitalis200 — APPROVED** [review]

**Pull request merged**